### PR TITLE
chore(version): bump algoliasearch version to v4.11.0

### PR DESCRIPTION
**Pull request by Haroenv**
* refactor(version): bump algoliasearch version to v0.11.0

* Import algoliasearch tarball package with type fix

* refactor(packages): Remove algoliasearch tarball

* refactor(types): Remove type work-arround for boundingBox

* Import algoliasearch client with type fix

* Remove unecessary type enforcement

* refactor(package): Bump algoliasearch to v0.11.0

fixes: #4916

testing the CI of #4938

### CODE REVIEW

**Comment by Haroenv** [comment]
@bidoubiwa, as we don't want CI errors with v3 of algoliasearch, I think this just bumping v4 wasn't enough to be able to remove the ts-expect-error, and they likely need to become ts-ignore instead now. In a next major version we can support TS only for v4, but at the moment we don't really enforce the major version. WDYT? Do you have a solution in mind?

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -163,6 +163,23 @@ To run the test continuously based on what you changed (useful when developing o
 yarn test --watch
 ```
 
+### Type checks
+
+Type checks ensure code is correctly typed both for code quality and compatibility for our TypeScript users.
+
+To ensure typing is correct in the latest version of `instantsearch.js`:
```
Comment (sarahdayan) [inline CONTRIBUTING.md:170]:
```suggestion
To ensure typing is correct in the latest version of InstantSearch.js, you can run the following command:

```

File: README.md [hunk 2]
```diff
@@ -102,6 +102,30 @@ npm install instantsearch.js algoliasearch
...
+To use InstantSearch in a TypeScript environment, depending on your `algoliasearch` version, different types must be imported:
+
+In case you are not using `algoliasearch` as the search client, it must still be installed in TypeScript environments.
+
+#### `algoliasearch` version 4
+
+V4 uses the types provided by both `algoliasearch` and `@algolia/client-search`.
+
+```bash
+yarn add algoliasearch@4 @algolia/client-search
```
Comment (sarahdayan) [inline README.md:116]:
```suggestion
yarn add algoliasearch@4 && yarn add @algolia/client-search -D
```

File: README.md [hunk 3]
```diff
@@ -102,6 +102,30 @@ npm install instantsearch.js algoliasearch
...
+```
+
+#### `algoliasearch` version 3
```
Comment (sarahdayan) [inline README.md:119]:
```suggestion
#### `algoliasearch` v3.x
```

File: README.md [hunk 4]
```diff
@@ -102,6 +102,30 @@ npm install instantsearch.js algoliasearch
...
+
+```bash
+npm install @types/algoliasearch@3
+# or
```
Comment (sarahdayan) [inline README.md:123]: Why do we document npm here and not for the other snippets?

File: README.md [hunk 5]
```diff
@@ -102,6 +102,30 @@ npm install instantsearch.js algoliasearch
...
+yarn add @types/algoliasearch@3
+```
+
+The Algoliasearch client version 3 support is close to an end and thus **deprecated**.
```
Comment (sarahdayan) [inline README.md:127]:
```suggestion
>v3.x is deprecated and will soon no longer be supported.
```

File: src/connectors/geo-search/__tests__/connectGeoSearch-test.ts [hunk 6]
```diff
@@ -660,7 +660,7 @@ See documentation: https://www.algolia.com/doc/api-reference/widgets/geo-search/
       }),
     ]);
 
-    // @ts-expect-error connector uses strings
+    // @ts-ignore
```
Comment (sarahdayan) [inline src/connectors/geo-search/__tests__/connectGeoSearch-test.ts:663]: Why finally ignoring?

**Review by sarahdayan — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (Haroenv) [reply README.md:116]: why dev dependency for that one? it's a "peer" (actually usually you won't need to rely on it manually as it's a dependency of algoliasearch), but a normal dependency would be fine imo

File: README.md (hunk 4 above)
Comment (Haroenv) [reply README.md:123]:
```suggestion
```

File: src/connectors/geo-search/__tests__/connectGeoSearch-test.ts (hunk 6 above)
Comment (Haroenv) [reply src/connectors/geo-search/__tests__/connectGeoSearch-test.ts:663]: it errors with v3 but doesn't with v4, if we do expect-error it won't pass on v4

File: README.md (hunk 2 above)
Comment (sarahdayan) [reply README.md:116]: I usually keep all non-runtime dependencies as dev dependencies. Not a blocker in front-end anyway, people can choose how they want they prefer.

Feel free to ignore.

**Pull request merged**